### PR TITLE
Publish behavior tree by default, update clients

### PR DESCRIPTION
**Pull request by fred-labs**
### Description
Instead of requesting a behavior tree snapshot publisher for every client, enable one by default. This can be used by e.g. clients or within a ros-bag

### Pull request checklist

- [ ] [Issue](https://github.com/IntelLabs/scenario_execution/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [x] [Docs](https://intellabs.github.io/scenario_execution/) reviewed and added / updated if needed (for bug fixes / features)
- [x] Format and Lint checks (`make check`) pass locally
- [x] PR applies Apache 2.0 License to all code files

### Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe)

### CODE REVIEW

File: docs/libraries.rst [hunk 1]
```diff
@@ -698,7 +698,9 @@ Wait for specific output in ROS log (i.e. ``/rosout`` topic). If any of the entr
 ``record_bag()``
 """"""""""""""""
 
-Record a ROS bag, stored in directory ``output_dir`` defined by command-line parameter (default: '.')
+Record a ROS bag, stored in directory ``output_dir`` defined by command-line parameter (default: '.').
+
+A common topic to record is ``/scenario_execution/snapshots`` which publishes changes within the behavior tree.
```
Comment (fmirus) [inline docs/libraries.rst:703]: Please add something like: "This is useful to visualize the scenario tree in rviz when replaying the bagfile"

File: simulation/gazebo/tb4_sim_scenario/launch/ignition_launch.py [hunk 2]
```diff
@@ -46,6 +47,8 @@ def generate_launch_description():
     pkg_irobot_create_ignition_plugins = get_package_share_directory(
         'irobot_create_ignition_plugins')
 
+    headless = LaunchConfiguration('headless')
+
```
Comment (fmirus) [inline simulation/gazebo/tb4_sim_scenario/launch/ignition_launch.py:51]: don't we need to add a `DeclareLaunchArgument` for headless, too?

File: scenario_execution_rviz/src/scenario_view.cpp [hunk 3]
```diff
@@ -93,39 +83,109 @@ int searchBehavior(const QString *child_id,
...
+    // expand everything
+    for (auto const &item : items) {
+      mScenarioView->expandItem(item);
+    }
+  } else {
+    QTreeWidgetItemIterator it(mScenarioView);
+    while (*it) {
+      auto msg_behavior = msg->behaviours.begin();
+      while (msg_behavior < msg->behaviours.end()) {
+        if ((*it)->data(2, 0) ==
```
Comment (fmirus) [inline scenario_execution_rviz/src/scenario_view.cpp:135]: do we need to cheack that `(*it)->data(2,0)` for these hard-coded values exists?

**Review by fmirus — CHANGES_REQUESTED** [review]

File: simulation/gazebo/tb4_sim_scenario/launch/ignition_launch.py (hunk 2 above)
Comment (fred-labs) [reply simulation/gazebo/tb4_sim_scenario/launch/ignition_launch.py:51]: it's already there.

**Review by fmirus — APPROVED** [review]

**Pull request merged**